tests/foreman/ui/test_host.py

```diff
@@ -1,14 +1,278 @@
+        cls.domain.organization = [cls.org]
+        cls.domain.dns = cls.proxy
+        cls.domain = cls.domain.update(['dns', 'location', 'organization'])
+
+        # Search if subnet is defined with given network.
+        # If so, just update its relevant fields otherwise,
+        # Create new subnet
+        network = '192.168.100.0'
+        cls.subnet = entities.Subnet().search(
+            query={u'search': u'network={0}'.format(network)})
+        if (len(cls.subnet) == 1):
+            cls.subnet = cls.subnet[0]
+            cls.subnet.domain = [cls.domain]
+            cls.subnet.location = [cls.loc]
+            cls.subnet.organization = [cls.org]
+            cls.subnet.dns = cls.proxy
+            cls.subnet.dhcp = cls.proxy
+            cls.subnet.tftp = cls.proxy
+            cls.subnet.discovery = cls.proxy
+            cls.subnet = cls.subnet.update(
+                ['domain',
+                 'discovery',
+                 'dhcp',
+                 'dns',
+                 'location',
+                 'organization',
+                 'tftp']
+            )
+        else:
+            # Create new subnet
+            cls.subnet = entities.Subnet(
+                name=gen_string('alpha'),
+                network=network,
+                mask=u'255.255.255.0',
+                domain=[cls.domain],
+                location=[cls.loc],
+                organization=[cls.org],
+                dns=cls.proxy,
+                dhcp=cls.proxy,
+                tftp=cls.proxy,
+                discovery=cls.proxy
+            ).create()
+
+        # Create Libvirt Compute-resource
+        cls.computeresource = entities.LibvirtComputeResource(
+            name=gen_string('alpha'),
+            provider=u'libvirt',
+            url=u'qemu+tcp://{0}:16509/system'.format(
+                conf.properties['main.server.hostname']),
+            set_console_password=False,
+            location=[cls.loc.id],
+            organization=[cls.org.id],
+        ).create()
+
+        # Get the Partition table ID
+        cls.ptable = entities.PartitionTable().search(
+            query={
+                u'search': u'name="Kickstart default"'
+            }
+        )[0]
+
+        # Get the OS ID
+        cls.os = entities.OperatingSystem().search(
+            query={u'search': u'name="RedHat" AND major="6" AND minor="7"'})[0]
+
+        # Get the Provisioning template_ID and update with OS, Org, Location
+        cls.provisioning_template = entities.ConfigTemplate().search(
+            query={
+                u'search': u'name="Satellite Kickstart Default"'
+            }
+        )[0]
+        cls.provisioning_template.operatingsystem = [cls.os]
+        cls.provisioning_template.organization = [cls.org]
+        cls.provisioning_template.location = [cls.loc]
+        cls.provisioning_template = cls.provisioning_template.update(
+            ['location',
+             'operatingsystem',
+             'organization']
+        )
```

Comment:
This can also be written like

```py
cls.provisioning_template = cls.provisioning_template.update([
            'location',
            'operatingsystem',
            'organization'
        ])
```